eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1787,13 +1785,18 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
   def handleCurrentFeerate(c: CurrentFeerates, d: HasCommitments) = {
     val networkFeeratePerKw = c.feeratesPerKw.feePerBlock(target = nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)
     val currentFeeratePerKw = d.commitments.localCommit.spec.feeratePerKw
-    d.commitments.localParams.isFunder match {
-      case true if Helpers.shouldUpdateFee(currentFeeratePerKw, networkFeeratePerKw, nodeParams.onChainFeeConf.updateFeeMinDiffRatio) =>
-        self ! CMD_UPDATE_FEE(networkFeeratePerKw, commit = true)
-        stay
-      case false if Helpers.isFeeDiffTooHigh(currentFeeratePerKw, networkFeeratePerKw, nodeParams.onChainFeeConf.maxFeerateMismatch) =>
-        handleLocalError(FeerateTooDifferent(d.channelId, localFeeratePerKw = networkFeeratePerKw, remoteFeeratePerKw = d.commitments.localCommit.spec.feeratePerKw), d, Some(c))
-      case _ => stay
+    val shouldUpdateFee = d.commitments.localParams.isFunder &&
+      Helpers.shouldUpdateFee(currentFeeratePerKw, networkFeeratePerKw, nodeParams.onChainFeeConf.updateFeeMinDiffRatio)
+    val shouldClose = !d.commitments.localParams.isFunder &&
+      Helpers.isFeeDiffTooHigh(networkFeeratePerKw, currentFeeratePerKw, nodeParams.onChainFeeConf.maxFeerateMismatch) &&
+      d.commitments.hasPendingOrProposedHtlcs // we close only if we have HTLCs potentially at risk
```

Comment:
Proposed htlcs are not at risk (they haven't been signed)

Reply:
With the current implementation, they might be. Because once they're proposed, they'll eventually get signed and added to the commitment without going through the feerate mismatch verification (which we only do in `sendAdd` and `receiveAdd`).

I think for simplicity sake that it's better to close right now, allowing us to check the feerate mismatch only in these two functions while being safe. Otherwise we'd need to check it in a lot more places which is more error-prone.